src/main/java/com/ust/sourcecourse/configuration/service/DBDataSourceService.java

```diff
@@ -69,17 +78,86 @@ private DBTable getDBTable(SourceTable sourceTable) {
 				.yoyCount(sourceTable.getYoyCount()).rowCount(sourceTable.getRowCount()).size(sourceTable.getSize())
 				.build();
 		List<DBTableColumn> dbTableColumns = sourceTable.getSourceColumns().stream()
-				.map(sourceColumn -> getDBTableColumn(sourceColumn)).toList();
+				.map(sourceColumn -> getDBTableColumn1(sourceColumn)).toList();
 		return DBTable.builder().uid(sourceTable.getUid()).tableName(sourceTable.getName())
 				.description(sourceTable.getDescription()).metadata(metadata).tags(sourceTable.getTags())
 				.columns(dbTableColumns).build();
 	}
 
-	private DBTableColumn getDBTableColumn(SourceColumn sourceColumn) {
+	private DBTableColumn getDBTableColumn1(SourceColumn sourceColumn) {
 		DBTableColumnMetadata metadata = DBTableColumnMetadata.builder().type(sourceColumn.getType())
 				.isPrimary(sourceColumn.isPrimary()).isNullable(sourceColumn.isNullable())
 				.isUnique(sourceColumn.isUnique()).defaultValue(sourceColumn.getDefaultValue()).build();
 		return DBTableColumn.builder().uid(sourceColumn.getUid()).name(sourceColumn.getName())
 				.description(sourceColumn.getDescription()).metadata(metadata).tags(sourceColumn.getTags()).build();
 	}
+
+	public List<String> getDescriptionAndTags(Long uid) {
+		SourceTable homePage = sourceTableRepository.findById(uid).orElseThrow();
+
+		return homePage.getTags();
+	}
+
+	public List<String> getColumnTags(Long uid) {
+		Optional<SourceColumn> column = sourceColumnRepository.findById(uid);
+		if (column.isPresent()) {
+			return column.get().getTags();
+		} else {
+			throw new EntityNotFoundException("SourceColumn with uid " + uid + " not found");
+		}
+	}
+
+	public List<String> getTableTags(Long uid) {
+		Optional<SourceTable> table = sourceTableRepository.findById(uid);
+		if (table.isPresent()) {
+			return table.get().getTags();
+		} else {
+			throw new EntityNotFoundException("SourceTable with uid " + uid + " not found");
+		}
+	}
+
+	public SourceColumn updateColumnDescriptionAndTags(Long uid, HomeRequest request) {
+		Optional<SourceColumn> column = sourceColumnRepository.findById(uid);
+		if (column.isPresent()) {
+			column.get().setTags(request.getTags());
```

Comment:
You need to update description as well. 
Also, this is an update. So existing tags should not be removed. Setting tags directly from request will remove existing tags. Also, we need to handle duplicates. 

You can refer this https://github.com/sourcecoursepoc/source-course-configuration/blob/dev/src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java#L128

Reply:
resolved